lib/ansible/modules/network/cloudengine/ce_snmp_community.py

```diff
@@ -0,0 +1,941 @@
+                    replace('xmlns="http://www.huawei.com/netconf/vrp"', "")
+
+                root = ElementTree.fromstring(xml_str)
+                community_info = root.findall("data/snmp/communitys/community")
+                if community_info:
+                    for tmp in community_info:
+                        tmp_dict = dict()
+                        for site in tmp:
+                            if site.tag in ["communityName", "accessRight", "aclNumber", "mibViewName"]:
+                                tmp_dict[site.tag] = site.text
+
+                        result["community_info"].append(tmp_dict)
+
+                if result["community_info"]:
+                    for tmp in result["community_info"]:
+                        if "communityName" in tmp.keys():
+                            need_cfg = True
+
+                        if "accessRight" in tmp.keys():
+                            if state == "present":
+                                if tmp["accessRight"] != access_right:
+                                    need_cfg = True
+                            else:
+                                if tmp["accessRight"] == access_right:
+                                    need_cfg = True
+
+                        if acl_number:
+                            if "aclNumber" in tmp.keys():
+                                if state == "present":
+                                    if tmp["aclNumber"] != acl_number:
+                                        need_cfg = True
+                                else:
+                                    if tmp["aclNumber"] == acl_number:
+                                        need_cfg = True
+
+                        if community_mib_view:
+                            if "mibViewName" in tmp.keys():
+                                if state == "present":
+                                    if tmp["mibViewName"] != community_mib_view:
+                                        need_cfg = True
+                                else:
+                                    if tmp["mibViewName"] == community_mib_view:
+                                        need_cfg = True
+
+        result["need_cfg"] = need_cfg
+        return result
+
+    def check_snmp_v3_group_args(self, **kwargs):
+        """ Check snmp v3 group args """
+
+        module = kwargs["module"]
+        result = dict()
+        need_cfg = False
+        result["group_info"] = []
+        state = module.params['state']
+        group_name = module.params['group_name']
+        security_level = module.params['security_level']
+        acl_number = module.params['acl_number']
+        read_view = module.params['read_view']
+        write_view = module.params['write_view']
+        notify_view = module.params['notify_view']
+
+        community_name = module.params['community_name']
+        access_right = module.params['access_right']
+
+        if group_name and not security_level:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```